src/uu/expr/src/syntax_tree.rs

```diff
@@ -197,6 +208,93 @@ const PRECEDENCE: &[&[(&str, BinOp)]] = &[
     &[(":", BinOp::String(StringOp::Match))],
 ];
 
+fn check_brace_content_and_matching(s: &str) -> BraceContent {
+    let mut chars = s.chars().peekable();
```

Comment:
It looks like the "peekable" functionality is never used, so I think it can be removed.
```suggestion
    let mut chars = s.chars();
```